Update hooks priority for `wp_generate_attachment_metadata`

thrijith opened the pull request:
- Update hooks priority for `wp_generate_attachment_metadata` to work with azure storage
- Move `classifai_generate_image_alt_tags_source_url` filter to helper function
- Use `get_modified_image_source_url` where rescanning is done local file


### Description of the Change

- ClassifAI is generating tags on `10` [priority](https://github.com/10up/classifai/blob/develop/includes/Classifai/Providers/Azure/ComputerVision.php#L68), while the storage plugin uploads image to storage blob and deletes locally on [priority](https://github.com/10up/windows-azure-storage/blob/develop/windows-azure-storage.php#L125-L127) `9`, by the time it is getting there to generate tags, the local image is gone and alt text is not generated, reducing priority of the hook will allow it to generate the text and then upload to storage.

- I found a related PR https://github.com/10up/classifai/pull/250 where the hook priority was increased due to similar issues, setting hook priority to be in following order `smart_crop` at `7` and `alt_text` at `8` should hopefully fix issues both issues.

- The above changes only fix issue when the image is being uploaded, it doesn't work when we try to generate/re-generate/rescan image tags from media modal for an existing image, adding the filter introduced in https://github.com/10up/classifai/pull/217 it other places will help with the same.

### Verification Process

- Tested on project that was facing this issue, applying these changes fixed the alt generation for images and image was successfully uploaded to Azure storage as well.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues

- Doesn't have one right now, happy to open one if needed.

thrijith committed:
Update hooks priority for `wp_generate_attachment_metadata` to work with azure storage
Move `classifai_generate_image_alt_tags_source_url` filter to helper function
Use `get_modified_image_source_url` where rescanning is done local file

dkotter commented:
@thrijith Thanks for this PR. Code looks good and worked fine in my testing. There is a failed test though, if you can look at that and get it fixed up?

thrijith committed:
Fix test case assertion and allow having tests with empty assertions

thrijith commented:
> @thrijith Thanks for this PR. Code looks good and worked fine in my testing. There is a failed test though, if you can look at that and get it fixed up?

Thanks for the review @dkotter, the test failure seems unrelated to the change, I've updated the failing case it seems to be passing now, also added a change to fix some other alerts for risky tests.
